### PR TITLE
removed polymorphism section and moved examples for discriminators up…

### LINKED ISSUE #309: Clarify that 'discriminator' and 'discriminatorValue' can only be defined for global type definitions
Due to its nature, RC2 should explicitly state that defining `discriminator` and `discriminatorValue` should only be available for global type definitions and not for any types that are declared inline.

```yaml
# valid
types:
   Device:
     type: Phone | Notebook
     discriminator: kind
```

```yaml
# invalid
application/json: 
   type: Phone | Notebook
   discriminator: kind
```

### LINKED ISSUE #330: Remove Polymorphism section in favour for a better description for `discriminator` and `discriminatorValue`
The current RC1 section about [Runtime Polymorphism (Discriminators)](https://github.com/raml-org/raml-spec/blob/raml-10/versions/raml-10/raml-10.md#runtime-polymorphism-discriminators) is actually not needed anymore in RC2. 

For RC2, it's enough to add a bit more description for `discriminator` and `discriminatorValue`. 

------------------------------

| Name | Type | Description | 
|:---------|:------:|:-----------------|
| discriminator | `string` | The value must correspond to a name of one of the properties that are defined inside a type declaration. The `discriminator` facet can not be defined for inline type declarations and should only be used for scalar-valued properties.
| discriminatorValue | `string`  | You may use `discriminatorValue` facet if a type also defined the `discriminator` facet. The value of this facet should be a valid value of the property with the name equal to the value of `discriminator` facet; and its being used to identify the declaring type. This value should be unique in the hierarchy of the type.<br/><br/>**Default:** the name of the type where the discriminator is applied to

### PR DESCRIPTION
**Pull request by sichvoge**
… to the object type section; as well as changed description for discriminator and discriminatorValue to close #330

This PR automatically closes #309 and #307 with the new introduced "Discriminator" chapter.

### CODE REVIEW

File: versions/raml-10/raml-10.md [hunk 1]
```diff
@@ -460,8 +460,8 @@ All types that have the built-in object type at the root of their inheritance tr
 | maxProperties? | The maximum number of properties allowed for instances of this type.
 | additionalProperties? | JSON schema style syntax for declaring maps. See section [Map Types](#map-types) for more information.
 | patternProperties? | JSON schema style syntax for declaring key restricted maps. See section [Map Types](#map-types) for more information.
-| discriminator? | Type property name to be used as discriminator, or boolean
-| discriminatorValue? | The value of discriminator for the type.
+| discriminator? | You may use the `discriminator` property to be able to discriminate the concrete type of an individual object at runtime when, for example, payloads contain ambiguous types (achieved via unions or inheritance). The value must correspond to a name of one of the `properties` that are defined inside a type declaration. The `discriminator` property can not be defined for inline type declarations and should only be used for scalar-valued properties. See an example on how to use `discriminator` [here](#using-discriminator-example).<br/><br/>**Default:** the name of the type
```
Comment (petrochenko-pavel-a) [inline versions/raml-10/raml-10.md]: If the type is suitable to AC then `discriminator` property is not needed

File: versions/raml-10/raml-10.md [hunk 2]
```diff
@@ -659,6 +659,74 @@ types:
         type: Method
 ```
 
+#### Using Discriminator Example
```
Comment (petrochenko-pavel-a) [inline versions/raml-10/raml-10.md]: I think we need some more intro wording here.

File: versions/raml-10/raml-10.md [hunk 3]
```diff
@@ -659,6 +659,74 @@ types:
...
+data:
+  - name: A User
+    userId: 111
+    kind: User
+  - name: An Employee
+    employeeId: 222
+    kind: Employee
+```
+
+You can also override the value of the discriminator for each individual concrete class. In the following example we will replace the default type discriminator values by their lowercase versions using `discriminatorValue`:
```
Comment (petrochenko-pavel-a) [inline versions/raml-10/raml-10.md]: You probably mean `discriminatorValue`

File: versions/raml-10/raml-10.md (hunk 1 above)
Comment (sichvoge) [inline versions/raml-10/raml-10.md]: That is why we have the default right? The automatic classification (AC) uses that, am I correct? And I don't think we should directly mention AC in this context. So what do you want me to change?

File: versions/raml-10/raml-10.md (hunk 3 above)
Comment (sichvoge) [inline versions/raml-10/raml-10.md]: Don't understand. What are you missing or referring to?

File: versions/raml-10/raml-10.md (hunk 2 above)
Comment (sichvoge) [inline versions/raml-10/raml-10.md]: What do you think about? That is not a separated "Discriminator" section and only illustrates an example of using it. Is the description for the specific properties not enough? Are you thinking about just giving an intro to the example?

File: versions/raml-10/raml-10.md (hunk 1 above)
Comment (petrochenko-pavel-a) [inline versions/raml-10/raml-10.md]: >And I don't think we should directly mention AC in this context.
Agree. I actually did not like this phrase:
>You may use the `discriminator` property to be able to discriminate the concrete type of an individual object at runtime.

Basically it leads you to opinion that you need to use it in any case. In fact I like description here pretty much:
https://github.com/mulesoft-labs/typesystem-docs/blame/master/typesystem.md#L1197

File: versions/raml-10/raml-10.md (hunk 2 above)
Comment (petrochenko-pavel-a) [inline versions/raml-10/raml-10.md]: yep, I do not actually know how to write it correctly. But this looks like a bit unexpected.

File: versions/raml-10/raml-10.md (hunk 3 above)
Comment (petrochenko-pavel-a) [inline versions/raml-10/raml-10.md]: >the value of the discriminator for each individual concrete class
In fact you are overriding `discriminatorValue`

Comment (sichvoge) [inline versions/raml-10/raml-10.md]: Then it was wrong in RC1 as well. Modifying it.

**Comment by petrochenko-pavel-a** [comment]
Looks good to me.

**Pull request merged**